src/tests/test_rng.cpp

```diff
@@ -100,23 +111,68 @@ class HMAC_DRBG_Tests : public Text_Based_Test
 
 BOTAN_REGISTER_TEST("hmac_drbg", HMAC_DRBG_Tests);
 
-class HMAC_DRBG_Reseed_Tests : public Test
+class HMAC_DRBG_Unit_Tests : public Test
    {
-   public:
-      HMAC_DRBG_Reseed_Tests() : Test() {}
+   private:
+      class Broken_Entropy_Source : public Botan::Entropy_Source
+         {
+         public:
+            std::string name() const override { return "Broken Entropy Source"; }
 
-      std::vector<Test::Result> run() override
+            size_t poll(Botan::RandomNumberGenerator&) override
+               {
+               throw Botan::Exception("polling not available");
+               }
+         };
+
+      class Insufficient_Entropy_Source : public Botan::Entropy_Source
+         {
+         public:
+            std::string name() const override { return "Insufficient Entropy Source"; }
+
+            size_t poll(Botan::RandomNumberGenerator&) override
+               {
+               return 0;
+               }
+         };
+
+      class Request_Counting_RNG : public Botan::RandomNumberGenerator
+         {
+         public:
+            Request_Counting_RNG() : m_randomize_count(0) {};
+
+            bool is_seeded() const override { return true; }
+
+            void clear() override {}
+
+            void randomize(byte[], size_t) override
+               {
+               m_randomize_count++;
+               }
+
+            void add_entropy(const byte[], size_t) override {}
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3AS1186)